app/src/main/java/tttTDD/Java/Game.java

```diff
@@ -0,0 +1,48 @@
+package tttTDD.Java;
+
+import tttTDD.Java.Interfaces.Player;
+
+public class Game {
+    Board board;
+    Player player1;
+    Player player2;
+    Player currentPlayer;
+    boolean gameCompleted;
+    String gameWinner;
+
+    public Game(GameConfiguration gameConfig) {
+        board = new Board(gameConfig.getBoardSizeConfiguration());
+        initializePlayers(gameConfig.isComputerPlaying());
+        currentPlayer = player1;
+        gameCompleted = false;
+    }
+    // humanPlayer.move(int) => return int
+    // list all the things that it does and all the things that it needs
+    public boolean performMove(int playerInput) {
+        if (!currentPlayer.isComputer()) {
+            board.updateBoard(playerInput, currentPlayer.getMarker());
+        } else {
+            board.updateBoard(board.getFirstAvailableMove(), currentPlayer.getMarker());
+        }
+        gameCompleted = gameHasWinner();
+        swap();
```

Comment:
Also discussed in IPM, the performMove cannot swap players if the last move did not take effect, or else you'll potentially have the same player take 2 consecutive moves.

Reply:
good job on this